### DISCUSSION TITLE
Export lineage to image file

### DISCUSSION THREAD

**Discussion by ovcharenko** [reata/sqllineage#458] [Q&A]
Hi,

Is there a way to export table or column lineage data to an image file without using the built-in web server? Ideally, it should be rendered in the same way. For example, I created a PoC, but no matter what layout I tried, it only resulted in a dot presentation.

```python
import matplotlib.pyplot as plt
import networkx as nx
from sqllineage.runner import LineageRunner


class LineageRunnerExtended(LineageRunner):
    def to_image(self, image_filename, format="svg"):
        self._eval()

        graph = self._sql_holder.column_lineage_graph
        plt.figure(figsize=(100, 100))
        pos = nx.spring_layout(graph)

        nx.draw(graph, pos, with_labels=True)

        plt.savefig(image_filename, format=format)
        plt.close()
```

**Comment by reata** [comment]
No, that's not supported. Current visualization solution is built upon JavaScript graph library cytoscapejs where web server is a must.

sqllineage v1.0.0 was used to built upon networkx, matplotlib and pygraphviz for table lineage visualization. But once we started column lineage work, they didn't give me the visualization I wanted, perhaps I was doing it in a wrong way. Here's the table lineage code for your reference: https://github.com/reata/sqllineage/blob/v1.0.0/sqllineage/drawing.py

Feel free to try with column lineage, I'd be very happy to see it working.
